### ISSUE TITLE
Form checkbox/radio/select - field mapped - custom properties not loading 

### ISSUE THREAD

**Issue by david-gap** [mautic/mautic#13542]
### Mautic Version

5.0.x series

### Way of installing

I downloaded a release from https://www.mautic.org/mautic-releases

### PHP version

8.1.27

### What happened?

After Updating Mautic from 4.4.11 to 5.0.2 there is an issue with the form fields checkbox, radio and select.
If the input field is mapped with a contact/company field and custom properties are defined, the assigned contact/company fields list choices are loading.

**Custom properties**
![image](https://github.com/mautic/mautic/assets/23354033/57b79ede-8c03-432e-b4a3-9cb9061e239d)

**Output**
![image](https://github.com/mautic/mautic/assets/23354033/66e1432d-79db-471d-93ee-02701f50e960)

### How can we reproduce this issue?

Step 1: Create a form
Step 2: Add field type radio
Step 3: Connect field with the contact title field
Step 4: Define custom properties
Step 5: See form output

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by EsdersGmbH** [comment]
This seems like a duplicate of #13306, and I can confirm this issue also on our runing Mauti 5.x instance.

**Comment by Lealuli** [comment]
Same trouble here, all forms created up to now have to updated manually to find a solution. I used this for privacy policy and now have to be quick to not get into legal trouble on the website.

**Comment by david-gap** [comment]
Since a lot of time has already passed, I went looking for the problem myself and was able to find something.

In the directory app > bundles > FormBundle > views > Field in the files group.html.twig (line 145) and select.html.twig (line 144), the self-defined items are overwritten by the field values without checking the field synchronisation.

I have temporarily solved the problem with the following statement:
`{%- if field.properties.syncList is defined and field.properties.syncList is not empty and field.properties.syncList -%}`

The if statement encloses 16 lines (the same in both files) and looks like this:
`      {%- if field.properties.syncList is defined and field.properties.syncList is not empty and field.properties.syncList -%}`
`        {%- if mappedField.properties.list is defined and mappedField.properties.list is not empty -%}`
`          {%- set parseList = mappedField.properties.list -%}`
`        {%- elseif 'boolean' == mappedFieldType -%}`
`          {%- set parseList = {`
`                  '0': mappedField.properties.no,`
`                  '1': mappedField.properties.yes,`
`          }-%}`
`        {%- elseif 'country' == mappedFieldType -%}`
`          {%- set list = leadFieldCountryChoices() -%}`
`        {%- elseif 'region' == mappedFieldType -%}`
`          {%- set list = leadFieldRegionChoices() -%}`
`        {%- elseif 'timezone' == mappedFieldType -%}`
`          {%- set list = leadFieldTimezonesChoices() -%}`
`        {%- elseif 'locale' == mappedFieldType -%}`
`          {%- set list = leadFieldLocaleChoices() -%}`
`        {%- endif -%}`
`      {%- endif -%}`

I hope this helps.

**Comment by RCheesley** [comment]
Thanks folks, I'm going to close the original issue report given there's more conversation happening here. @david-gap would you be open to making a PR for that? Here's some [docs](https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests), as it's a bug fix please use the 5.1 branch.
